src/compiler/checker.ts

```diff
@@ -9694,6 +9687,13 @@ namespace ts {
             return spread;
         }
 
+        function isSpreadableProperty(prop: Symbol): boolean {
+            // We approximate own properties as non-methods plus methods that are inside the object literal
+            return prop.flags & (SymbolFlags.Method | SymbolFlags.GetAccessor)
+                ? !prop.declarations.some(decl => isClassLike(decl.parent))
+                : !(prop.flags & SymbolFlags.SetAccessor); // Setter without getter is not spreadable

```

Comment:
A Setter is actually spreadable and results in `undefined`:

```js
({
    foo: 1,
    ...{set foo(v) {}}
}).foo === undefined;
```

Reply:
Will do. It seemed to me like this condition was added in this PR

Reply:
No, I thought so too, but it's just adding accessors to the list of things that are skipped when they come from a class declaration.